src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TestServiceContainerWeakRefPass.php

```diff
@@ -32,6 +32,10 @@ public function process(ContainerBuilder $container)
 
         foreach ($definitions as $id => $definition) {
             if ($id && '.' !== $id[0] && (!$definition->isPublic() || $definition->isPrivate()) && !$definition->getErrors() && !$definition->isAbstract()) {
+                if (!$definition->isShared()) {
+                    $definition->setShared(true);
```

Comment:
wouldn't this cause issues regarding inlining ? Rules are not the same for shared and not shared services.

Reply:
It will change the resulting inlining yes that's the goal, but inlining is only an optimization: it has no external side effect, so this is neutral, behavior wise.